docs/datasources/softlayer_ssh_key.md

```diff
@@ -0,0 +1,41 @@
+# `softlayer_ssh_key`
+
+Use this data source to import the details of an *existing* SSH key as a read-only data source.
+
+## Example Usage
+
+```hcl
+data "softlayer_ssh_key" "public_key" {
+    label = "Terraform Public Key"
+}
+```
+
+The fields of the data source can then be referenced by other resources within the same configuration using
+interpolation syntax. For example, when specifying SSH keys in a softlayer_virtual_guest resource configuration,
+the numeric "IDs" are often unknown. Using the above data source as an example, it would be possible to
+reference the `id` property in a softlayer_virtual_guest resource:
+
+```hcl
+resource "softlayer_virtual_guest" "vm1" {
+    ...
+    ssh_keys = ["${data.softlayer_ssh_key.public_key.id}"]
```

Comment:
Good question - I have never been able to get it to work with quotes omitted.  Here's what I get when I use this (quoted) syntax:

```
ssh_keys.#:               "1"
ssh_keys.0:               "383111"
```

So clearly, string.  But `ssh_keys = [383111]` yields the exact same result.  So it seems like Terraform treats everything as string and type asserts as needed.